components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -268,137 +263,114 @@ func (w *NotebookWebhook) Handle(ctx context.Context, req admission.Request) adm
 	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledNotebook)
 }
 
-func (w *NotebookWebhook) ClusterWideProxyIsEnabled() bool {
-	proxyResourceList := &configv1.ProxyList{}
-	err := w.Client.List(context.TODO(), proxyResourceList)
-	if err != nil {
-		return false
-	}
-
-	for _, proxy := range proxyResourceList.Items {
-		if proxy.Name == "cluster" {
-			if proxy.Status.HTTPProxy != "" && proxy.Status.HTTPSProxy != "" &&
-				proxy.Status.NoProxy != "" {
-				// Update Proxy Env variables map
-				proxyEnvVars["HTTP_PROXY"] = proxy.Status.HTTPProxy
-				proxyEnvVars["HTTPS_PROXY"] = proxy.Status.HTTPSProxy
-				proxyEnvVars["NO_PROXY"] = proxy.Status.NoProxy
-				if proxy.Spec.TrustedCA.Name != "" {
-					proxyEnvVars["PIP_CERT"] = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
-					proxyEnvVars["REQUESTS_CA_BUNDLE"] = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
-					proxyEnvVars["SSL_CERT_FILE"] = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
-				}
-				return true
-			}
-		}
-	}
-	return false
-
-}
-
 // InjectDecoder injects the decoder.
 func (w *NotebookWebhook) InjectDecoder(d *admission.Decoder) error {
 	w.Decoder = d
 	return nil
 }
 
-func InjectProxyConfig(notebook *nbv1.Notebook) error {
-	notebookContainers := &notebook.Spec.Template.Spec.Containers
-	var imgContainer corev1.Container
+func CheckAndMountCACertBundle(ctx context.Context, cli client.Client, notebook *nbv1.Notebook) error {
 
-	// Add trusted-ca volume
-	notebookVolumes := &notebook.Spec.Template.Spec.Volumes
-	certVolumeExists := false
-	certVolume := corev1.Volume{
-		Name: "trusted-ca",
-		VolumeSource: corev1.VolumeSource{
-			ConfigMap: &corev1.ConfigMapVolumeSource{
-				LocalObjectReference: corev1.LocalObjectReference{
-					Name: "trusted-ca",
-				},
-				Items: []corev1.KeyToPath{{
-					Key:  "ca-bundle.crt",
-					Path: "tls-ca-bundle.pem",
-				},
-				},
-			},
-		},
-	}
-	for index, volume := range *notebookVolumes {
-		if volume.Name == "trusted-ca" {
-			(*notebookVolumes)[index] = certVolume
-			certVolumeExists = true
-			break
-		}
-	}
-	if !certVolumeExists {
-		*notebookVolumes = append(*notebookVolumes, certVolume)
+	configMapName := "odh-trusted-ca-bundle"
+
+	// Fetch the list of ConfigMaps from the cluster
+	configMapList := &corev1.ConfigMapList{}
+	if err := cli.List(ctx, configMapList); err != nil {
+		return err
```

Comment:
I changed the logic of looking for the odh-trusted-ca in all permitted namespaces.
Now, is checking only inside the given namespace https://github.com/atheo89/kubeflow/blob/ca-cert/components/odh-notebook-controller/controllers/notebook_webhook.go#L278 
In case that the `odh-trusted-ca-bundle` not exist inside the datascience project namespace, it throws the following error.
![image](https://github.com/opendatahub-io/kubeflow/assets/42587738/7496c7c7-09fa-452f-8d7c-ca69093a4fa2)

Reply:
instead of the throwing the error, we shoudl just log this in logging 
as the cert are optionally mounted.

Reply:
fixed